client/web/web.go

```diff
@@ -162,12 +162,16 @@ func NewServer(opts ServerOpts) (s *Server, err error) {
 		opts.LocalClient = &tailscale.LocalClient{}
 	}
 	s = &Server{
-		mode:        opts.Mode,
-		logf:        opts.Logf,
-		devMode:     envknob.Bool("TS_DEBUG_WEB_CLIENT_DEV"),
-		lc:          opts.LocalClient,
-		cgiMode:     opts.CGIMode,
-		pathPrefix:  opts.PathPrefix,
+		mode:    opts.Mode,
+		logf:    opts.Logf,
+		devMode: envknob.Bool("TS_DEBUG_WEB_CLIENT_DEV"),
+		lc:      opts.LocalClient,
+		cgiMode: opts.CGIMode,
+		// In enforcePrefix, we add the necessary leading '/'. If we did not
+		// strip 1 or more leading '/'s here, we would end up redirecting
+		// clients to e.g. //example.com (a schema-less URL that points to
+		// another site). See https://github.com/tailscale/corp/issues/16268.
+		pathPrefix:  strings.TrimPrefix(opts.PathPrefix, "/"),
```

Comment:
this only strips a single leading slash, so I think you'd still have problems with `///example.com`.  Maybe we pass this through `path.Clean`, which will collapse all contiguous slashes to a single slash, as well as remove `../` segments ?

Reply:
Yeah, turns out `TrimLeft` removes *all* leading elements that are in the `cutset`.

I'm also thinking about any code anywhere (Windows, browsers, ...) being tolerant of '\' and treating it like '/', so how about this:

```go
strings.TrimLeft(path.Clean(opts.PathPrefix), "/\\")
```

?

Reply:
probably overkill, but yeah that works 😄